xgettext.js

```diff
@@ -123,30 +132,29 @@ XGettext.prototype._normalizeKeyword = function( keyword ) {
  */
 XGettext.prototype._parseInput = function( input ) {
 	var comments = [],
-		parseOptions = { locations: true },
+		parseOptions = this.options.parseOptions,
 		ast;
 
+	ast = parser.parse( input, parseOptions );
+
 	if ( typeof this.options.commentPrefix !== 'undefined' ) {
 		// Optionally locate translator comments
 		var rxCommentMatch = new RegExp( '^\\s*' + this.options.commentPrefix, 'i' );
-
-		parseOptions.onComment = function( block, text, start, end, line ) {
+		for ( var i = 0, l = ast.comments.length; i < l; i++ ) {
```

Comment:
Safe and more readable to use [`Array#forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) I think.